pkg/engine/systemd.go

```diff
@@ -12,7 +17,57 @@ func systemdPodman(ctx context.Context, mo *FileMountOptions) error {
 	if err := fileTransferPodman(ctx, mo); err != nil {
 		return fmt.Errorf("Repo: %s, Method: %s, %v", err)
 	}
-	// TODO: Add logic to start services, root/non-root
-	klog.Infof("Repo: %s, systemd target successfully processed", mo.Target.Name)
+	sd := mo.Target.Systemd
+	if !sd.Enable {
+		klog.Infof("Repo: %s, systemd target successfully processed", mo.Target.Name)
+		return nil
+	}
+	return enableSystemdService(mo.Conn, sd.Root, mo.Dest, filepath.Base(mo.Path), mo.Target.Name)
+}
+
+func enableSystemdService(conn context.Context, root bool, systemdPath, service, repoName string) error {
+	klog.Infof("Identifying if systemd image exists locally")
+	if err := fetchImage(conn, systemdImage); err != nil {
+		return err
+	}
+	os.Setenv("ROOT", "true")
+	if !root {
+		//os.Setenv("ROOT", "false")
+		klog.Info("At this time, harpoon non-root user cannot enable systemd service on the host")
+		klog.Infof("To enable this non-root service, run 'systemctl --user enable %s' on host machine", service)
+		klog.Info("To enable service as root, run with Systemd.Root = true")
+		return nil
+	}
+
+	os.Setenv("SERVICE", service)
+	s := specgen.NewSpecGenerator(systemdImage, false)
+	runMount := "/run/systemd"
+	if !root {
```

Comment:
We may need an enhancement request for figuring out the uid dynamically for the user but it should not stop this PR from merging